src/ImmersedBoundaries/mask_immersed_field.jl

```diff
@@ -89,6 +89,55 @@ end
     @inbounds field[i, j, k] = scalar_mask(i, j, k, grid, grid.immersed_boundary, loc..., value, field, mask)
 end
 
+#####
+##### Masking a `ReducedField`
+#####
+
+# We mask a `ReducedField` if the entire reduced direction is immersed.
+# This requires a sweep over the reduced direction
+
+function mask_immersed_field!(field::ReducedField, grid::ImmersedBoundaryGrid, loc, value)
+    loc  = instantiate.(loc)
+    dims = reduced_dimensions(field)
+    launch!(architecture(field), grid, size(field), _mask_immersed_reduced_field!, field, dims, loc, grid, value)
+    return nothing
+end
+
+@kernel function _mask_immersed_reduced_field!(field, dims, loc, grid, value)
+    i, j, k = @index(Global, NTuple)
+    mask = mask_reduced_direction(i, j, k, grid, dims, loc)
+    @inbounds field[i, j, k] = ifelse(mask, value, field[i, j, k]) 
+end
+
+@inline masking_range(i, grid, dim, dims) = ifelse(dim ∈ dims, 1:size(grid, dim), i:i)
+
+@inline function mask_reduced_direction(i₀, j₀, k₀, grid, dims, loc)
```

Comment:
```suggestion
@inline function inactive_dimensions(i₀, j₀, k₀, grid, dims, loc)
```

This function returns `true` if the entire `dims` are inactive. Thus it is like `inactive`, but applied to a whole dimension.

Reply:
also the argument can contain multiple dimensions so you need plural